airflow/models/renderedtifields.py

```diff
@@ -236,14 +232,15 @@ def _do_delete_old_records(
         session: Session,
     ) -> None:
         # This query might deadlock occasionally and it should be retried if fails (see decorator)
+        # breakpoint()
         stmt = (
             delete(cls)
             .where(
                 cls.dag_id == dag_id,
                 cls.task_id == task_id,
                 tuple_not_in_condition(
                     (cls.dag_id, cls.task_id, cls.run_id),
-                    session.query(ti_clause.c.dag_id, ti_clause.c.task_id, ti_clause.c.run_id),
+                    select(ti_clause.c.dag_id, ti_clause.c.task_id, ti_clause.c.run_id),
```

Comment:
Damn, `tuple_not_in_condition` and `tuple_in_condition` do not work well with the new style. Let’s leave this out, I’ll try to convert them in a separate PR.

Reply:
it's failing specifically for MSSQL, for rest all databases it is working fine. Will check today if we need to have specific logic for that